src/com/pentalog/pentastagiu/homework2/HomeworkApp.java

```diff
@@ -1,7 +1,338 @@
+                            basket.setTotalPrice(basket.getTotalPrice() + candy2.getPrice()*quantity);
+                            candy2.setQuantity(candy2.getQuantity()-quantity);
+                            selectedGoodQuantity = true;
+                        }
+                        else
+                            System.out.println("We do not have enough candy2");
+                    }
+            }
+
+            System.out.println("Do you want to continue?(y/n)");
+            Scanner in2 = new Scanner(System.in);
+            String wantToContinue = in2.nextLine();
+            if(wantToContinue.equals("n"))
+                doYouWantToContinue = false;
+        }
+
+        double firstPrice;
+        switch (customer.getMembership()) {
+            case "SILVER ":
+                firstPrice = ((double)basket.getTotalPrice()-(10*basket.getTotalPrice()/100));
+                System.out.println("10% discount. Total price " + ((double)basket.getTotalPrice()-(10*basket.getTotalPrice()/100)));
+                break;
+            case "GOLD ":
+                firstPrice = ((double)basket.getTotalPrice()-(20*basket.getTotalPrice()/100));
+                System.out.println("20% discount. Total price " + ((double)basket.getTotalPrice()-(20*basket.getTotalPrice()/100)));
+                break;
+            default:
+                firstPrice = ((double)basket.getTotalPrice());
+                System.out.println("0% discount. Total price " + ((double)basket.getTotalPrice()));
+                break;
+        }
+
+        customer.setMembership("SILVER");
+        System.out.println("You became a silver member! Use that in your advantage");
+
+        /*doYouWantToContinue = true;
+        while(doYouWantToContinue) {
+            System.out.println("Select a product to buy: book1, book2, book3, candy1, candy2");
+            String choice = in.nextLine();
+
+            switch (choice) {
+                case "book1":
+                    basket.setTotalPrice(basket.getTotalPrice() + book1.getPrice());
+                    break;
+                case "book2":
+                    basket.setTotalPrice(basket.getTotalPrice() + book2.getPrice());
+                    break;
+                case "book3":
+                    basket.setTotalPrice(basket.getTotalPrice() + book3.getPrice());
+                    break;
+                case "candy1":
+                    basket.setTotalPrice(basket.getTotalPrice() + candy1.getPrice());
+                    break;
+                case "candy2":
+                    basket.setTotalPrice(basket.getTotalPrice() + candy2.getPrice());
+                    break;
+            }
+
+            System.out.println("Do you want to continue?(y/n)");
+            String wantToContinue = in.nextLine();
+            if(wantToContinue.equals("n"))
+                doYouWantToContinue = false;
+        }*/
+
+        double secondPrice;
+        switch (customer.getMembership()) {
```

Comment:
urmatoarele 10 linii de cod sunt identice cu liniile 136-149 (cu exceptia numelui variabilei) --> cand vezi ca logica se repeta, refactorizeaza, extrage codul intr-o metoda (de ex calculatePriceWithDiscount) pentru ca e mai usor de citit si mai clar + mult mai usor de schimbat(daca cineva iti cerea sa schimbi discountul care se aplica pentru membership GOLD, trebuia sa vii si sa cauti pe peste tot prin cod pe unde l-ai folosit; o alta sugestue: pastreaza discountul strans legat de menbership: printr-un enum in care sa ai un field discount)